lib/adapters/parsers/grok.js

```diff
@@ -0,0 +1,76 @@
+var base = require('./base');
+var errors = require('../../errors');
+var grok = require('node-grok');
+var Promise = require('bluebird');
+var readlines = require('readlines');
+
+module.exports = base.extend({
+
+    initialize: function(options) {
+        this.pattern = options.pattern || '%{GREEDYDATA:message}';
+    },
+
+    parseStream: function(stream, emit) {
+        var self = this;
+
+        return new Promise(function(resolve, reject) {
+            grok.loadDefault(function(patterns) {
+                var buffer = [];
+                var liner = readlines.readlineStream();
```

Comment:
Seems to me like we should use the builtin node.js implementation of reading a file line by line instead of pulling in this half-baked module:
https://nodejs.org/api/readline.html#readline_example_read_file_stream_line_by_line